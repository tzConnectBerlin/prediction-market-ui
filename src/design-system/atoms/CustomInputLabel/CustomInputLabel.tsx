import styled from '@emotion/styled';
import { InputLabel, InputLabelProps, TextFieldProps, Theme } from '@material-ui/core';

interface StyledInputLabelProps extends InputLabelProps {
  /**
   * Material-UI Theme
   */
  theme?: Theme;
  /**
   * Font weight to provide (default: 600)
   */
  fontWeight?: number | string;
  /**
   * Margin top (default: -12%)
   */
  marginTop?: number | string;
  /**
   * Custom class to add to asterisk (default: label-asterisk)
   */
  asteriskClass?: string;
}
export interface CustomInputLabelProps extends StyledInputLabelProps {
  /**
   * Show asterisk or not
   */
  required?: boolean;
  /**
   * Text to show
   */
  label: TextFieldProps['label'];
}

const StyledInputLabel = styled(InputLabel)<StyledInputLabelProps>`
  margin-top: ${({ marginTop }) => marginTop ?? '-12%'};
  font-weight: ${({ fontWeight }) => fontWeight ?? 600};
  & .label-asterisk {
    vertical-align: sub;
  }
  color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
`;

export const CustomInputLabel: React.FC<CustomInputLabelProps> = ({
  label,
  required,
  asteriskClass = 'label-asterisk',
  ...rest
}) => {
  return (
    <StyledInputLabel required={required} classes={{ asterisk: asteriskClass }} {...rest}>
      {label}
    </StyledInputLabel>
  );
};
