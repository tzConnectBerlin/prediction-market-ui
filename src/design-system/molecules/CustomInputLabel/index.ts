export type { CustomInputLabelProps } from './CustomInputLabel';
export type { CustomInputChipProps } from './CustomInputLabel';
export { CustomInputLabel, CustomInputLabel as default } from './CustomInputLabel';
