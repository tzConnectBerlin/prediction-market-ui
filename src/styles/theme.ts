import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
