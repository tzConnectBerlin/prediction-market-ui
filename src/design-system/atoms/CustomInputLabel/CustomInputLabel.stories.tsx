import { Story, Meta } from '@storybook/react/types-6-0';
import { ThemeProvider } from '@material-ui/core/styles';
import { theme } from '../../../theme';
import { CustomInputLabel, CustomInputLabelProps } from './CustomInputLabel';

export default {
  title: 'Atom/CustomInputLabel',
  component: CustomInputLabel,
} as Meta;

const Template: Story<CustomInputLabelProps> = (args) => (
  <ThemeProvider theme={theme}>
    <CustomInputLabel {...args} />
  </ThemeProvider>
);

export const Default = Template.bind({});
Default.args = {
  label: 'Enter a question',
  theme,
  marginTop: '2%',
};

export const Required = Template.bind({});
Required.args = {
  label: 'Enter a question',
  theme,
  marginTop: '2%',
  required: true,
};

export const WithoutTheme = Template.bind({});
WithoutTheme.args = {
  label: 'Enter a question',
  marginTop: '2%',
  required: true,
};
