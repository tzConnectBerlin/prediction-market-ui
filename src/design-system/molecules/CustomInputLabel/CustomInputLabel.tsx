import styled from '@emotion/styled';
import { InputLabel, InputLabelProps, TextFieldProps, Theme, Grid } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import { CustomTooltip, CustomTooltipProps } from '../../atoms/CustomTooltip/CustomTooltip';

interface StyledInputLabelProps extends InputLabelProps {
  /**
   * Material-UI Theme
   */
  theme?: Theme;
  /**
   * Font weight to provide (default: 600)
   */
  fontWeight?: number | string;
  /**
   * Margin top (default: -12%)
   */
  marginTop?: number | string;
  /**
   * Custom class to add to asterisk (default: label-asterisk)
   */
  asteriskClass?: string;
}
export interface CustomInputLabelProps extends StyledInputLabelProps {
  /**
   * Show asterisk or not
   */
  required?: boolean;
  /**
   * Text to show
   */
  label: TextFieldProps['label'];
  tooltipProps?: CustomTooltipProps;
}

const StyledInputLabel = styled(InputLabel)<StyledInputLabelProps>`
  margin-top: ${({ marginTop }) => marginTop ?? '-12%'};
  font-weight: ${({ fontWeight }) => fontWeight ?? 600};
  color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
  &.Mui-disabled {
    color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
    opacity: 0.38;
  }
  & .label-asterisk {
    vertical-align: sub;
  }
`;

export const CustomInputLabel: React.FC<CustomInputLabelProps> = ({
  label,
  required,
  asteriskClass = 'label-asterisk',
  disabled,
  tooltipProps,
  ...rest
}) => {
  const theme = useTheme();
  return (
    <Grid container>
      <Grid item xs={11}>
        <StyledInputLabel
          required={required}
          classes={{ asterisk: asteriskClass }}
          theme={theme}
          disabled={disabled}
          {...rest}
        >
          {label}
        </StyledInputLabel>
      </Grid>
      {tooltipProps && (
        <Grid xs={1} item marginTop={rest.marginTop ?? '-12%'}>
          <CustomTooltip {...tooltipProps} />
        </Grid>
      )}
    </Grid>
  );
};
