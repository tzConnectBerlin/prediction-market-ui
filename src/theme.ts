import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
    light: 'rgba(212, 241, 204, 0.12)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.04)',
    '300': 'rgba(29, 34, 39, 0.03)',
    '200': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.54)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.54)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
  // action: {
  //   disabledBackground: "#0166FF",
  //   disabled: "rgba(255,255,255,0.3)",
  // },
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
});

export const darkTheme = createTheme({
  palette: {
    ...lightThemePalette,
    primary: {
      dark: 'rgba(214, 231, 255, 1)',
      main: 'rgba(204, 224, 255, 1)',
      light: 'rrgba(204, 224, 255, .12)',
      contrastText: 'rgba(29, 34, 39, 0.87)',
    },
    secondary: {
      dark: 'rgba(204, 224, 255, 0.14)',
      main: 'rgba(204, 224, 255, 0.12)',
      light: 'rgba(204, 224, 255, 0.14)',
      contrastText: 'rgba(204, 224, 255, 1)',
    },
    tertiary: {
      dark: 'rgba(212, 241, 204, 1)',
      main: 'rgba(212, 241, 204, 1)',
      light: 'rgba(212, 241, 204, 0.12)',
    },
    grey: {
      '900': 'rgba(255, 255, 255, 1)',
      '800': 'rgba(255, 255, 255, 0.87)',
      '700': 'rgba(255, 255, 255, 0.38)',
      '600': 'rgba(255, 255, 255, 0.2)',
      '500': 'rgba(255, 255, 255, 0.11)',
      '400': 'rgba(255, 255, 255, 0.09)',
      '300': 'rgba(255, 255, 255, 0.03)',
      '200': 'rgba(255, 255, 255, 0.02)',
    },
    text: {
      primary: 'rgba(255, 255, 255, 0.87)',
      secondary: 'rgba(255, 255, 255, 0.54)',
      disabled: 'rgba(255, 255, 255, 0.38)',
    },
    buttonText: {
      primary: 'rgba(29, 34, 39, 0.87)',
      secondary: 'rgba(29, 34, 39, 0.54)',
      disabled: 'rgba(29, 34, 39, 0.38)',
    },
    background: {
      default: 'rgba(29, 34, 39, 1)',
      paper: 'rgba(45, 49, 54, 1)',
    },
    mode: 'dark',
  },
  zIndex: {
    tooltip: 9999,
  },
});
