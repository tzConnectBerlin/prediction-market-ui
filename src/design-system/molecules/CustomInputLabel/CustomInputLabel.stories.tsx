import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { CustomInputLabel, CustomInputLabelProps } from './CustomInputLabel';

export default {
  title: 'Molecule/CustomInputLabel',
  component: CustomInputLabel,
} as Meta;

const Template: Story<CustomInputLabelProps> = (args) => <CustomInputLabel {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'Enter a question',
};

export const Required = Template.bind({});
Required.args = {
  label: 'Enter a question',
  required: true,
};
