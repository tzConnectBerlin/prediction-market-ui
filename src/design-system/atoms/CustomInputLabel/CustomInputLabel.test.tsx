import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react';
import { theme } from '../../../theme';
import { CustomInputLabelProps, CustomInputLabel } from './CustomInputLabel';

const onlyRequiredProps: CustomInputLabelProps = {
  label: 'Create a question',
};

const defaultProps: CustomInputLabelProps = {
  ...onlyRequiredProps,
  marginTop: '1%',
  theme,
};

describe('Snapshot testing CustomInputLabel Component', () => {
  it('renders correctly with default props', () => {
    const inputLabel = renderer.create(<CustomInputLabel {...defaultProps} />).toJSON();
    expect(inputLabel).toMatchSnapshot();
  });

  it('renders correctly when theme not provided', () => {
    const inputLabel = renderer.create(<CustomInputLabel {...onlyRequiredProps} />).toJSON();
    expect(inputLabel).toMatchSnapshot();
  });
});

describe('Element testing CustomInputLabel Component', () => {
  it('render correctly CustomInputLabel with default props', async () => {
    const { findByText } = render(<CustomInputLabel {...defaultProps} />);

    waitFor(() => {
      expect(findByText(/Create a question/i)).toBeInTheDocument();
    });
  });

  it('render correctly CustomInputLabel with only required props', async () => {
    const { findByText } = render(<CustomInputLabel {...defaultProps} />);

    waitFor(() => {
      expect(findByText(/Create a question/i)).toBeInTheDocument();
    });
  });
});
