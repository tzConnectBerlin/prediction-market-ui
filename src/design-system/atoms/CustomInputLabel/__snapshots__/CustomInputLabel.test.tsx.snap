// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing CustomInputLabel Component renders correctly when theme not provided 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-11im5ym-MuiFormLabel-root-MuiInputLabel-root"
>
  Create a question
</label>
`;

exports[`Snapshot testing CustomInputLabel Component renders correctly with default props 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-1q183x4-MuiFormLabel-root-MuiInputLabel-root"
  marginTop="1%"
>
  Create a question
</label>
`;
